### PR TITLE
Less Overpowered Commando ERTs (Marsoc, WY Whiteout, UPP Commandos)

### PR DESCRIPTION
**Pull request by Steelpoint**
# About the pull request

This PR adds three new alternative Commando distress calls for admin/staff only usage. These are variants of the Marsoc, WY Whiteout and UPP Commando that use less overpowered gear. 

The non-OP versions have no unique identifier in the admin menus, however the overpowered versions are denoted with the affix of "(!DEATHSQUAD!)". This applies to both the distress call and the preset.

Marsoc: Have lost HEAP/Special Explosives/Unique Medical Items
WY Whiteout: Have lost HEAP/Special OP Flamer/Extra OP Smartgun, they are still combat Synthetics
UPP: Have lost their cloaking abilities, they still retain a cloak for the aesthetic but it is non-functional.

# Explain why it's good for the game

Commando ERTs are virtually never used in game outside of zombies or for clean-up after an event due to how oppressively overpowered they are. 

This addition will allow admins/staff to make usage of these unique ERTs more commonly for events without having to worry about them annihilating the entire server. While each ERT is still quite powerful, they are not as destructive. 

I am 100% open to further changes as needed. 

# Testing Photographs and Procedure
<details>
<summary>Screenshots & Videos</summary>

Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.

</details>

# Changelog
:cl:
add: Less powerful variants of the Commando ERTs (Marsoc, WY Whiteout, UPP Commandos) have been added for admin use.
admin: The powerful versions of commando ERTs are denoted with (!DEATHSQUAD!) in the distress call and equipment preset menus. 
/:cl:

### CODE REVIEW

**Comment by Steelpoint** [comment]
> > The regular, overpowered, ERTs are denoted with the affix of "(!WHITEOUT!)".
> 
> This seems like it'll be confusing, "Whiteout" is the WY thing, it wouldn't make sense for Marsoc / UPP to have this affix, why not just say it's a "Deathsquad"?

That does sound better than what I thought, done.

File: code/datums/emergency_calls/deathsquad.dm [hunk 1]
```diff
@@ -49,10 +49,57 @@
 
 	addtimer(CALLBACK(GLOBAL_PROC, GLOBAL_PROC_REF(to_chat), H, SPAN_BOLD("Objectives: [objectives]")), 1 SECONDS)
 
+/datum/emergency_call/death_lowthreat
```
Comment (BeagleGaming1) [inline code/datums/emergency_calls/deathsquad.dm:52]: Why not make the lowthreat variants a subtype of the regular one, and just /create_member with weaker gear?

File: code/datums/emergency_calls/deathsquad.dm [hunk 2]
```diff
@@ -107,3 +154,31 @@
...
+
+/datum/emergency_call/marsoc_lowthreat
+	name = "Marine Raider Operatives (No Overpowered Gear)"
+	mob_max = 8
+	mob_min = 5
+	probability = 0
+	shuttle_id = "Distress_PMC"
+	name_of_spawn = /obj/effect/landmark/ert_spawns/distress_pmc
+
+/datum/emergency_call/marsoc_lowthreat/create_member(datum/mind/M)
```
Comment (BeagleGaming1) [inline code/datums/emergency_calls/deathsquad.dm:166]: Single letter var

File: code/datums/emergency_calls/deathsquad.dm [hunk 3]
```diff
@@ -49,10 +49,57 @@
...
+
+
+// DEATH SQUAD--------------------------------------------------------------------------------
+/datum/emergency_call/death_lowthreat/create_member(datum/mind/M, turf/override_spawn_loc)
+	var/turf/spawn_loc = override_spawn_loc ? override_spawn_loc : get_spawn_point()
+
+	if(!istype(spawn_loc))
+		return //Didn't find a useable spawn point.
+
+	var/mob/living/carbon/human/H = new(spawn_loc)
```
Comment (BeagleGaming1) [inline code/datums/emergency_calls/deathsquad.dm:74]: Single letter var

File: code/datums/emergency_calls/upp_commando.dm [hunk 4]
```diff
@@ -51,3 +51,53 @@
...
+	name = "UPP Commandos (No Overpowered Gear)"
+	mob_max = 6
+	probability = 0
+	objectives = "Stealthily assault the ship. Use your silenced weapons, tranquilizers, and night vision to get the advantage on the enemy. Take out the power systems, comms and engine. Stick together and keep a low profile."
+	shuttle_id = "Distress_UPP"
+	name_of_spawn = /obj/effect/landmark/ert_spawns/distress_upp
+	item_spawn = /obj/effect/landmark/ert_spawns/distress_upp/item
+	hostility = TRUE
+
+/datum/emergency_call/upp_commando/print_backstory(mob/living/carbon/human/M)
```
Comment (BeagleGaming1) [inline code/datums/emergency_calls/upp_commando.dm:64]: Single letter var

File: code/datums/emergency_calls/upp_commando.dm [hunk 5]
```diff
@@ -51,3 +51,53 @@
...
+	to_chat(M, SPAN_NOTICE(" This allows you to speak privately with your fellow UPP allies."))
+	to_chat(M, SPAN_NOTICE(" Utilize it with your radio to prevent enemy radio interceptions."))
+
+/datum/emergency_call/upp_commando/create_member(datum/mind/M, turf/override_spawn_loc)
+	var/turf/spawn_loc = override_spawn_loc ? override_spawn_loc : get_spawn_point()
+
+	if(!istype(spawn_loc))
+		return //Didn't find a useable spawn point.
+
+	var/mob/living/carbon/human/H = new(spawn_loc)
```
Comment (BeagleGaming1) [inline code/datums/emergency_calls/upp_commando.dm:87]: Single letter var

File: code/modules/gear_presets/whiteout.dm [hunk 6]
```diff
@@ -216,3 +216,182 @@
...
+
+
+	//face
+	new_human.equip_to_slot_or_del(new /obj/item/clothing/glasses/night/m42_night_goggles/m42c, WEAR_EYES)
+	new_human.equip_to_slot_or_del(new /obj/item/clothing/mask/gas/pmc/leader, WEAR_FACE)
+	new_human.equip_to_slot_or_del(new /obj/item/device/radio/headset/distress/commando, WEAR_L_EAR)
+	//head
+	new_human.equip_to_slot_or_del(new /obj/item/clothing/head/helmet/marine/veteran/pmc/commando, WEAR_HEAD)
+	//uniform
+	var/obj/item/clothing/under/marine/veteran/pmc/commando/M = new()
```
Comment (BeagleGaming1) [inline code/modules/gear_presets/whiteout.dm:246]: Single letter var

File: code/modules/gear_presets/whiteout.dm [hunk 7]
```diff
@@ -216,3 +216,182 @@
...
+	new_human.equip_to_slot_or_del(new /obj/item/device/defibrillator/upgraded, WEAR_IN_BACK)
+	new_human.equip_to_slot_or_del(new /obj/item/roller, WEAR_IN_BACK)
+	//face
+	new_human.equip_to_slot_or_del(new /obj/item/clothing/glasses/night/medhud, WEAR_EYES)
+	new_human.equip_to_slot_or_del(new /obj/item/clothing/mask/gas/pmc/leader, WEAR_FACE)
+	new_human.equip_to_slot_or_del(new /obj/item/device/radio/headset/distress/commando, WEAR_L_EAR)
+	//head
+	new_human.equip_to_slot_or_del(new /obj/item/clothing/head/helmet/marine/veteran/pmc/commando, WEAR_HEAD)
+	//uniform
+	var/obj/item/clothing/under/marine/veteran/pmc/commando/M = new()
```
Comment (BeagleGaming1) [inline code/modules/gear_presets/whiteout.dm:293]: Single letter var

File: code/modules/gear_presets/whiteout.dm [hunk 8]
```diff
@@ -216,3 +216,182 @@
...
+	new_human.equip_to_slot_or_del(new /obj/item/explosive/plastic/breaching_charge, WEAR_IN_BACK)
+	new_human.equip_to_slot_or_del(new /obj/item/storage/firstaid/synth, WEAR_IN_BACK)
+	//face
+	new_human.equip_to_slot_or_del(new /obj/item/clothing/glasses/night/m42_night_goggles/m42c, WEAR_EYES)
+	new_human.equip_to_slot_or_del(new /obj/item/clothing/mask/gas/pmc/leader, WEAR_FACE)
+	new_human.equip_to_slot_or_del(new /obj/item/device/radio/headset/distress/commando, WEAR_L_EAR)
+	//head
+	new_human.equip_to_slot_or_del(new /obj/item/clothing/head/helmet/marine/veteran/pmc/commando, WEAR_HEAD)
+	//uniform
+	var/obj/item/clothing/under/marine/veteran/pmc/commando/M = new()
```
Comment (BeagleGaming1) [inline code/modules/gear_presets/whiteout.dm:375]: Single letter var

**Review by BeagleGaming1 — CHANGES_REQUESTED** [review]

**Comment by Steelpoint** [comment]
Tidied up the code. 

References to the overpowered versions of the commandos are now all properly denoted with !DEATHSQUD! whereas the less powerful versions have no unique identifier. This applies to both the distress call and the preset equipment.

File: code/datums/emergency_calls/upp_commando.dm [hunk 9]
```diff
@@ -51,3 +51,53 @@
...
+	name = "UPP Commandos"
+	mob_max = 6
+	probability = 0
+	objectives = "Stealthily assault the ship. Use your silenced weapons, tranquilizers, and night vision to get the advantage on the enemy. Take out the power systems, comms and engine. Stick together and keep a low profile."
+	shuttle_id = "Distress_UPP"
+	name_of_spawn = /obj/effect/landmark/ert_spawns/distress_upp
+	item_spawn = /obj/effect/landmark/ert_spawns/distress_upp/item
+	hostility = TRUE
+
+/datum/emergency_call/upp_commando/print_backstory(mob/living/carbon/human/mind)
```
Comment (morrowwolf) [inline code/datums/emergency_calls/upp_commando.dm:64]: Accidentally doubled looks like

File: code/datums/emergency_calls/upp_commando.dm [hunk 10]
```diff
@@ -51,3 +51,53 @@
 
 	addtimer(CALLBACK(GLOBAL_PROC, GLOBAL_PROC_REF(to_chat), H, SPAN_BOLD("Objectives: [objectives]")), 1 SECONDS)
 
+/datum/emergency_call/upp_commando/lowthreat
```
Comment (morrowwolf) [inline code/datums/emergency_calls/upp_commando.dm:54]: All of the typepaths that have "lowthreat" need to be snake_case'd so "low_threat"

File: code/datums/emergency_calls/upp_commando.dm [hunk 11]
```diff
@@ -51,3 +51,53 @@
...
+	name = "UPP Commandos"
+	mob_max = 6
```
Comment (morrowwolf) [inline code/datums/emergency_calls/upp_commando.dm:56]: Any of the vars that are the same here as the parent need to be removed. So like /datum/emergency_call/upp_commando already sets the mob_max to 6 and this inherits that so it does not need to be set here. Do a check for these dupes across the changes please.

**Review by morrowwolf — CHANGES_REQUESTED** [review]

**Comment by Steelpoint** [comment]
I believe all requested changes have been made.

**Review by harryob — APPROVED** [review]

**Pull request merged**